pkg/inst/worker/daemon.R

```diff
@@ -0,0 +1,57 @@
+# Worker daemon
+
+rLibDir <- Sys.getenv("SPARKR_RLIBDIR")
+script <- paste(rLibDir, "SparkR/worker/worker.R", sep = "/")
+
+# preload SparkR package, speedup worker
+.libPaths(c(rLibDir, .libPaths()))
+suppressPackageStartupMessages(library(SparkR))
+
+port <- as.integer(Sys.getenv("SPARKR_WORKER_PORT"))
+inputCon <- socketConnection(port = port, open = "rb", blocking = TRUE, timeout = 3600)
+
+# Read from connection, retrying to read exactly 'size' bytes
+readBinFull <- function(con, size) {
+  # readBin() could be interruptted by signal, we have no way to tell
+  # if it's interruptted or the socket is closed, so we retry at most
+  # 20 times, to avoid the deadloop if the socket is closed
+  c <- 1
```

Comment:
When the socket is opened in blocking mode, readBin() should read enough bytes util: 1) socket is closed; 2) timeout, We can verify these two, and it's not be interrupted when I `kill -CHLD PID` manually, you are right. 

I ran into issues that readInt() in daemon will return 0 (length is 0) when the child process die, after that readInt() will block until it got 4 bytes. What's the reason readBin() will return without enough bytes? (the socket is still open, the timeout is big enough).